### ISSUE TITLE
Prune jobs not using `--host` in restic command

### ISSUE THREAD

**Issue by shreddedbacon** [k8up-io/k8up#686]
### Description

Prune jobs don't appear to use the `--host` parameter when pruning backups, according to restic this is supported https://restic.readthedocs.io/en/latest/060_forget.html#removing-snapshots-according-to-a-policy

Backups appear to be created using the `--host` parameter

If two different namespaces back up to the same bucket, but have different retention periods applied to the schedules in each namespace, then it seems that all backups in that bucket would be pruned with both retention policies?

### Additional Context

Looking at the following snapshots in a bucket, the `host` is here.
```
/ # restic snapshots
repository e3fb2a8b opened successfully, password is correct
created new cache in /root/.cache/restic
ID        Time                 Host                       Tags        Paths
---------------------------------------------------------------------------------------------------------------------------------
2d53a3e8  2022-04-29 23:55:42  example-test                    /example-test-solr-persistent.solr.tar
2e46c96c  2022-05-01 20:44:39  example-production              /example-production-mariadb-prebackuppod.mariadb.sql
15a34f84  2022-05-01 20:44:41  example-production              /example-production-solr-persistent.solr.tar
d2e0272f  2022-05-01 20:44:42  example-production              /data/nginx
21769fcb  2022-05-01 23:23:39  example-master                  /example-master-mariadb-prebackuppod.mariadb.sql
6c228d9d  2022-05-01 23:23:40  example-master                  /example-master-solr-persistent.solr.tar
f284b55f  2022-05-01 23:23:42  example-master                  /data/nginx
```

But if we had the following schedules in each of the namespaces `example-production|master|test` then would they all be pruned to the shortest retention period, meaning the `master|production` host based snapshots would also be pruned?

(forgive the v1alpha1 api, we are looking to update, but this is still relevant to current k8up codebase after looking through the code)
One daily, 1 weekly, 1 monthly retention on the test environment:
```
kind: Schedule
apiVersion: backup.appuio.ch/v1alpha1
metadata:
  name: schedule1
  namespace: example-test
spec:
  backend:
    <snip>
  backup:
    <snip>
  check:
    <snip>
  prune:
    resources: {}
    retention:
      keepDaily: 1
      keepHourly: 0
      keepMonthly: 1
      keepWeekly: 1
    schedule: 50 5 * * 6
```

7 daily, 6 weekly, 1 monthly retention on the other environments:
```
kind: Schedule
apiVersion: backup.appuio.ch/v1alpha1
metadata:
  name: schedule2
  namespace: example-production
spec:
  backend:
    <snip>
  backup:
    <snip>
  check:
    <snip>
  prune:
    resources: {}
    retention:
      keepDaily: 7
      keepHourly: 0
      keepMonthly: 1
      keepWeekly: 6
    schedule: 50 5 * * 6
---
kind: Schedule
apiVersion: backup.appuio.ch/v1alpha1
metadata:
  name: schedule1
  namespace: example-master
spec:
  backend:
    <snip>
  backup:
    <snip>
  check:
    <snip>
  prune:
    resources: {}
    retention:
      keepDaily: 7
      keepHourly: 0
      keepMonthly: 1
      keepWeekly: 6
    schedule: 50 5 * * 6
```

### Expected Behavior

Prune should use the `host` field when pruning to prevent other snapshots being pruned also

### Implementation ideas

From the linked restic docs:
> Additionally, you can restrict the policy to only process snapshots which have a particular hostname with the `--host` parameter, or tags with the `--tag` option. 

We just need to add the parameter to the restic CLI for prunes: https://github.com/k8up-io/k8up/blob/master/restic/cli/prune.go#L16

```go
args := []string{
	"--prune",
	"--host",
	cfg.Config.Hostname,
}
```

### Version of K8up

1.2.0

### Version of Kubernetes

v1.21

### Distribution of Kubernetes

EKS

**Comment by shreddedbacon** [comment]
We just verified that this is the case on v1.2.0 of k8up, and reviewing the code in this repo for the latest version it appears to be the case here too.

**Comment by Kidswiss** [comment]
Hi @shreddedbacon 

Thanks for submitting this bug!

Indeed the prune is currently not limited by namespace via the `--host` parameter.

I'm amending your text with some implementation ideas to fix it.

**Comment by ccremer** [comment]
Hi @shreddedbacon 
I'm about to add this parameter to the `restic forget` invocation.
However, please note that K8up v1.x is not supported anymore, and as such the fix is only applied to v2.x

**Comment by shreddedbacon** [comment]
Totally expected this would be the case, we're looking to update soon so won't be a problem hopefully :crossed_fingers:

**Comment by ccremer** [comment]
Released the fix as v2.3.4. Thanks for reporting!
